test/Atmos/Parameterizations/Microphysics/KM_PySDMachine/test/utils/KM_PySDM.jl

```diff
@@ -0,0 +1,273 @@
+
+    builder.add_dynamic(displacement)
+
+    builder.add_dynamic(pkg_dynamics.Coalescence(kernel = pysdm.config.kernel))
+
+
+    attributes = environment.init_attributes(
+        spatial_discretisation = pkg_init.spatial_sampling.Pseudorandom(),
+        spectral_discretisation = pkg_init.spectral_sampling.ConstantMultiplicity(
+            spectrum = pysdm.config.spectrum_per_mass_of_dry_air,
+        ),
+        kappa = pysdm.config.kappa,
+    )
+
+
+    pkg_PySDM_products = pyimport("PySDM.products")
+    pysdm_products = []
+    push!(
+        pysdm_products,
+        pkg_PySDM_products.WaterMixingRatio(
+            name = "qc",
+            description_prefix = "liquid",
+            radius_range = (0.0, Inf),
+        ),
+    )
+
+    push!(pysdm_products, pkg_PySDM_products.RelativeHumidity())
+    push!(pysdm_products, pkg_PySDM_products.ParticleMeanRadius())
+    push!(pysdm_products, pkg_PySDM_products.PeakSupersaturation())
+    push!(pysdm_products, pkg_PySDM_products.ActivatingRate())
+    push!(pysdm_products, pkg_PySDM_products.DeactivatingRate())
+    push!(pysdm_products, pkg_PySDM_products.CondensationTimestepMin())
+    push!(pysdm_products, pkg_PySDM_products.CondensationTimestepMax())
+    push!(
+        pysdm_products,
+        pkg_PySDM_products.CloudDropletEffectiveRadius(
+            radius_range = (0.0, Inf),
+        ),
+    )
+
+    pysdm.particulator = builder.build(attributes, products = pysdm_products)
+
+    displacement.upload_courant_field(courant_field)
+
+    pysdm.exporter = pkg_vtk_exp.VTKExporter(verbose = true)
+
+    return nothing
+end
+
+
+function do_step!(pysdm, varvals, t)
+
+    dynamics = pysdm.particulator.dynamics
+
+    dynamics["Displacement"]()
+
+    update_pysdm_fields!(pysdm, varvals, t)
+
+    pysdm.particulator.env.sync()
+
+    dynamics["Condensation"]()
+
+    pysdm.particulator._notify_observers()
+
+    export_attributes_to_vtk(pysdm)
+    export_products_to_vtk(pysdm)
+
+    pysdm.particulator.n_steps += 1
+end
+
+
+function update_pysdm_fields!(pysdm, vals, t)
+
+    liquid_water_mixing_ratio = pysdm.particulator.products["qc"].get() * 1e-3
+
+    liquid_water_specific_humidity = liquid_water_mixing_ratio
+
+    q_tot = vals["q_tot"][:, 1, :]
+    q_tot = bilinear_interpol(q_tot)
+
+    q = THDS.PhasePartition.(q_tot, liquid_water_specific_humidity, 0.0)
+
+    qv = THDS.vapor_specific_humidity.(q)
+
+    e_int = vals["e_int"][:, 1, :]
+    e_int = bilinear_interpol(e_int)
+
+    T = THDS.air_temperature.(param_set, e_int, q)
+
+    ρ = pysdm.rhod
+    thd = THDS.dry_pottemp.(param_set, T, ρ)
+
+    pysdm.particulator.env.set_thd(thd)
+    pysdm.particulator.env.set_qv(qv)
+
+    return nothing
+end
+
+function fini!(pysdm, varvals, t)
+    write_pvd(pysdm)
+end
+
+function bilinear_interpol(A)
+    A = [(A[y, x - 1] + A[y, x]) / 2 for y in 1:size(A)[1], x in 2:size(A)[2]]
+    A = [(A[y - 1, x] + A[y, x]) / 2 for y in 2:size(A)[1], x in 1:size(A)[2]]
+    return A
+end
```

Comment:
The name of the function was misleading, because what it does is the specific case of interpolation at the cell center, given values from ClimateMachine at cell corners. We have changed it to `cell_center_mean` so it better reflects the logic (average of 4 nearest neighbors). https://github.com/CliMA/ClimateMachine.jl/pull/2244/commits/7e7f454d6647475b92b912205fd081f8cd88e569